x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/schemas.test.ts

```diff
@@ -0,0 +1,1195 @@
+          references: ['index-1'],
+          query: 'some query',
+          language: 'kuery',
+          max_signals: 0,
+        }).error
+      ).toBeTruthy();
+    });
+
+    test('max_signals can be 1', () => {
+      expect(
+        updateSignalSchema.validate<Partial<SignalAlertParamsRest>>({
+          id: 'rule-1',
+          description: 'some description',
+          from: 'now-5m',
+          to: 'now',
+          index: ['index-1'],
+          name: 'some-name',
+          severity: 'severity',
+          interval: '5m',
+          type: 'query',
+          references: ['index-1'],
+          query: 'some query',
+          language: 'kuery',
+          max_signals: 1,
+        }).error
+      ).toBeFalsy();
+    });
+  });
+
+  describe('find signals schema', () => {
+    test('empty objects do validate', () => {
+      expect(findSignalsSchema.validate<Partial<FindParamsRest>>({}).error).toBeFalsy();
+    });
+
+    test('all values validate', () => {
+      expect(
+        findSignalsSchema.validate<Partial<FindParamsRest>>({
+          per_page: 5,
+          page: 1,
+          sort_field: 'some field',
+          fields: ['field 1', 'field 2'],
+        }).error
+      ).toBeFalsy();
+    });
+
+    test('made up parameters do not validate', () => {
+      expect(
+        findSignalsSchema.validate<Partial<FindParamsRest & { madeUp: string }>>({
+          madeUp: 'hi',
+        }).error
+      ).toBeTruthy();
+    });
+
+    test('per_page validates', () => {
+      expect(
+        findSignalsSchema.validate<Partial<FindParamsRest>>({ per_page: 5 }).error
+      ).toBeFalsy();
+    });
+
+    test('page validates', () => {
+      expect(findSignalsSchema.validate<Partial<FindParamsRest>>({ page: 5 }).error).toBeFalsy();
+    });
+
+    test('sort_field validates', () => {
+      expect(
+        findSignalsSchema.validate<Partial<FindParamsRest>>({ sort_field: 'some value' }).error
+      ).toBeFalsy();
+    });
+
+    test('fields validates with a string', () => {
+      expect(
+        findSignalsSchema.validate<Partial<FindParamsRest>>({ fields: ['some value'] }).error
+      ).toBeFalsy();
+    });
+
+    test('fields validates with multiple strings', () => {
+      expect(
+        findSignalsSchema.validate<Partial<FindParamsRest>>({
+          fields: ['some value 1', 'some value 2'],
+        }).error
+      ).toBeFalsy();
+    });
+
+    test('fields does not validate with a number', () => {
+      expect(
+        findSignalsSchema.validate<Partial<Omit<FindParamsRest, 'fields'>> & { fields: number[] }>({
+          fields: [5],
+        }).error
+      ).toBeTruthy();
+    });
+
+    test('per page has a default of 20', () => {
+      expect(findSignalsSchema.validate<Partial<FindParamsRest>>({}).value.per_page).toEqual(20);
+    });
+
+    test('page has a default of 1', () => {
+      expect(findSignalsSchema.validate<Partial<FindParamsRest>>({}).value.page).toEqual(1);
+    });
+  });
+});
```

Comment:
Expand this file if github is trying to hide it as this is all important tests for the system.